### PR TITLE
 Added Support to ECC certificates

### PR DESCRIPTION
**Pull request by fmonteiro93**
- Removed unnecessary guards to allow ECC keys to be used in addition to RSA keys. No other change was necessary in order to support ECC certificates.
- The CA unit tests were extended to feature ECC certificates
- Extended CSR unit tests to use ECC keys
- Extended X509/verification tests  to use ECC key certificates

### CODE REVIEW

**Review by FabioEstevesSousa — APPROVED** [review]
Looks OK.

**Review by FabioEstevesSousa — APPROVED** [review]
This commit looks good to.

File: tests/unit/test_ca.cpp [hunk 1]
```diff
@@ -46,6 +48,8 @@ class CATest : public ::testing::Test
 
     std::unique_ptr<X509Certificate> _rootCert;
```
Comment (sergiizubchevskyy) [inline tests/unit/test_ca.cpp]: Since you added EccCert, maybe rename _rootCert to _rootRsaCert? Improves reading.

File: tests/unit/test_ca.cpp [hunk 2]
```diff
@@ -56,13 +60,17 @@ class CATest : public ::testing::Test
 
     std::unique_ptr<CertificateAuthority> _ca;
```
Comment (sergiizubchevskyy) [inline tests/unit/test_ca.cpp]: Same comment as above.

**Review by pmvfonseca — APPROVED** [review]
Looks fine to me!

File: tests/unit/test_x509.cpp [hunk 3]
```diff
@@ -70,6 +74,12 @@ class X509Test : public ::testing::Test
     std::unique_ptr<X509Certificate> _year9999;
 
     std::unique_ptr<AsymmetricPublicKey> _root1_pubkey;
+
+
```
Comment (sergiizubchevskyy) [inline tests/unit/test_x509.cpp]: Delete new line :smiling_imp:

File: tests/unit/test_x509.cpp [hunk 4]
```diff
@@ -329,6 +344,88 @@ BAMCA0gAMEUCIFMm751uiLYek33gkcLHyCMdXntcwXUdgoEtOuq04Yr7AiEAk62k
 -----END CERTIFICATE-----)"
 };
 
+
```
Comment (sergiizubchevskyy) [inline tests/unit/test_x509.cpp]: One new line above the limit :stuck_out_tongue_closed_eyes:

File: tests/unit/test_x509.cpp [hunk 5]
```diff
@@ -490,11 +587,22 @@ TEST_F(X509Test, test9999EdgeCaseAsn1DateLong)
 TEST_F(X509Test, testIfPubkeyIsCorrectlyExtractedFromCertificate)
 {
     EXPECT_EQ(_root1->getPublicKey(), *_root1_pubkey);
+
```
Comment (sergiizubchevskyy) [inline tests/unit/test_x509.cpp:589]: Delete extra new line inside "TEST_F(X509Test, test1970EdgeCaseAsn1DateShort)" :+1:

File: tests/unit/test_x509.cpp [hunk 6]
```diff
@@ -329,6 +344,88 @@ BAMCA0gAMEUCIFMm751uiLYek33gkcLHyCMdXntcwXUdgoEtOuq04Yr7AiEAk62k
 -----END CERTIFICATE-----)"
 };
 
+
+const std::string X509Test::_eccRootKeyPem{
+        R"(-----BEGIN PRIVATE KEY-----
```
Comment (sergiizubchevskyy) [inline tests/unit/test_x509.cpp]: Maybe harmonize the formatting among all certificates and keys?

**Review by sergiizubchevskyy — CHANGES_REQUESTED** [review]

File: tests/unit/test_ca.cpp (hunk 1 above)
Comment (fmonteiro93) [reply tests/unit/test_ca.cpp]: Done

File: tests/unit/test_ca.cpp (hunk 2 above)
Comment (fmonteiro93) [reply tests/unit/test_ca.cpp]: Done

File: tests/unit/test_x509.cpp (hunk 3 above)
Comment (fmonteiro93) [reply tests/unit/test_x509.cpp]: Done!!

File: tests/unit/test_x509.cpp (hunk 4 above)
Comment (fmonteiro93) [reply tests/unit/test_x509.cpp]: Done

File: tests/unit/test_x509.cpp (hunk 6 above)
Comment (fmonteiro93) [reply tests/unit/test_x509.cpp]: Agreed

File: tests/unit/test_x509.cpp (hunk 5 above)
Comment (fmonteiro93) [reply tests/unit/test_x509.cpp:589]: Done

**Review by sergiizubchevskyy — APPROVED** [review]
Good job :100:

**Pull request merged**